scripts/mkstatic.py

```diff
@@ -1263,6 +1316,140 @@ def write(self, metadata_request, metadata_define, arguments, debug):
+                                    else:
+                                        raise Exception("THIS SHOULD NOT HAPPEN WITH CAPGEN'S METADATA PARSER")
+
+                                    # DH* TODO REMOVE THIS ENTIRE BLOCK in a future PR to feature/capgen
+                                    # This block should not be needed, the metadata parser should take care
+                                    # of flagging invalid dimensions for the host model or the physics.
+                                    # TODO: create a test suite to make sure these things are caught
+                                    # by the metadata parser - do this on the feature/capgen branch!
+                                    if ccpp_stage == 'run':
+                                        # This should not happen when parsing metadata with capgen's metadata parser, remove?
+                                        if dims[1] == CCPP_HORIZONTAL_LOOP_EXTENT and not dim0:
+                                            raise Exception(f"Invalid metadata for scheme {scheme_name}: " + \
+                                                            f"horizontal dimension for {var_standard_name} is {var.dimensions}")
+                                        # This should not happen when parsing metadata with capgen's metadata parser, remove?
+                                        elif CCPP_HORIZONTAL_LOOP_BEGIN in dims or CCPP_HORIZONTAL_LOOP_END in dims or \
+                                                CCPP_HORIZONTAL_DIMENSION in dims:
+                                            raise Exception(f"Invalid metadata for scheme {scheme_name}: " + \
+                                                            f"horizontal dimension for {var_standard_name} is {var.dimensions}")
+                                    else:
+                                        # This should not happen when parsing metadata with capgen's metadata parser, remove?
+                                        if dims[1] == CCPP_HORIZONTAL_DIMENSION and not dim0:
+                                            raise Exception(f"Invalid metadata for scheme {scheme_name}: " + \
+                                                            f"horizontal dimension for {var_standard_name} is {var.dimensions}")
+                                        # This should not happen when parsing metadata with capgen's metadata parser, remove?
+                                        if CCPP_HORIZONTAL_LOOP_BEGIN in dims or CCPP_HORIZONTAL_LOOP_END in dims or \
+                                                CCPP_LOOP_EXTENT in dims:
+                                            raise Exception(f"Invalid metadata for scheme {scheme_name}: " + \
+                                                            f"horizontal dimension for {var_standard_name} is {var.dimensions}")
+                                    # *DH
+
+                                # DH* TODO: WE CANNOT ACTIVATE USING EXPLICIT HORIZONTAL DIMENSIONS
```

Comment:
Same as above, we need to fix the way we handle inactive variables to make the switch to contiguous arrays.